File: 25304/25304.cpp
/* ************************************************************************** */
/*                                                                            */
/*                                                        :::      ::::::::   */
/*   25304.cpp                                          :+:      :+:    :+:   */
/*                                                    +:+ +:+         +:+     */
/*   By: jihoolee <jihoolee@student.42SEOUL.kr>     +#+  +:+       +#+        */
/*                                                +#+#+#+#+#+   +#+           */
/*   Created: 2022/08/07 18:25:50 by jihoolee          #+#    #+#             */
/*   Updated: 2022/08/07 18:31:07 by jihoolee         ###   ########.fr       */
/*                                                                            */
/* ************************************************************************** */

#include <iostream>

int main(void) {
  int total_price;
  int product_cnt;
  int price;
  int quantity;
  int sum = 0;

  std::cin >> total_price;
  std::cin >> product_cnt;
  for (int i = 0; i < product_cnt; ++i) {
    std::cin >> price >> quantity;
    sum += price * quantity;
  }
  if (total_price == sum)
    std::cout << "Yes";
  else
    std::cout << "No";
  return 0;
}

File: 25304/25304.py
# **************************************************************************** #
#                                                                              #
#                                                         :::      ::::::::    #
#    25304.py                                           :+:      :+:    :+:    #
#                                                     +:+ +:+         +:+      #
#    By: jihoolee <jihoolee@student.42SEOUL.kr>     +#+  +:+       +#+         #
#                                                 +#+#+#+#+#+   +#+            #
#    Created: 2022/08/07 18:32:17 by jihoolee          #+#    #+#              #
#    Updated: 2022/08/07 18:42:31 by jihoolee         ###   ########.fr        #
#                                                                              #
# **************************************************************************** #

def mul(iterable):
    result = 1

    for elem in iterable:
        result *= elem;
    return result

def main():
    total_price = int(input())
    product_cnt = int(input())
    if (sum([mul(list(map(int, input().split()))) for _ in range(product_cnt)]) == total_price):
        print('Yes')
    else:
        print('No')


if (__name__ == "__main__"):
    main()
